queries.sql

```diff
@@ -732,8 +732,209 @@ FROM
+  AND date_of_visit between DATE '2020-04-01'
+  AND '2020-08-30'
+ORDER BY
+  date_of_visit DESC;
+
+--  visited_date |  visitor
+-- --------------+------------
+--  2020-07-22   | Boarmon
+--  2020-07-22   | Angemon
+--  2020-07-22   | Devimon
+--  2020-07-22   | Plantmon
+--  2020-07-22   | Gabumon
+--  2020-07-22   | Agumon
+--  2020-05-24   | Blossom
+--  2020-05-24   | Squirtle
+--  2020-05-24   | Charmander
+--  2020-05-24   | Pikachu
+-- (10 rows)
+------------------------------------------------------------------------------
+/* What animal has the most visits to vets? */
+SELECT
+  species.name AS animal,
+  count(*) AS most_VISITS
+FROM
+  visits
+  JOIN species ON animal_id = species.id
+GROUP BY
+  species.name
+ORDER BY
+  most_VISITS DESC
+LIMIT
+  1;
+
+--  animal  | most_visits
+-- ---------+-------------
+--  Digimon |          13
+-- (1 row)
+------------------------------------------------------------------------------
+/*Who was Maisy Smith's first visit?*/
+SELECT
+  species.name AS visited,
+  vets.name AS vet
+FROM
+  visits
+  JOIN species ON animal_id = species.id
+  JOIN vets ON vet_id = vets.id
+WHERE
+  vets.name = 'Maisy Smith'
+ORDER BY
+  date_of_visit ASC
+LIMIT
+  1;
+
+--  visited |     vet
+-- ---------+-------------
+--  Digimon | Maisy Smith
+-- (1 row)
+------------------------------------------------------------------------------------
+/* Details for most recent visit: animal information, vet information, and date of visit */
+SELECT
+  vets.name AS vet_name,
+  species.name AS specialist_of,
+  animals.name AS animal_name,
+  species.name AS animal_type,
+  visits.date_of_visit
+from
+  visits
+  JOIN animals ON animal_id = animals.species_id
+  JOIN vets ON vet_id = vets.id
+  JOIN species ON animal_id = species.id
+ORDER BY
+  date_of_visit ASC
+LIMIT
+  1;
+
+--   vet_name   | specialist_of | animal_name | animal_type | date_of_visit
+-- -------------+---------------+-------------+-------------+---------------
+--  Maisy Smith | Digimon       | Agumon      | Digimon     | 2019-01-24
+-- (1 row)
+--------------------------------------------------------------------------------------
+/* How many visits were with a vet that did not specialize in that animal's species? */
+SELECT
+  count(*)
+FROM
+  visits
+  JOIN specializations ON visits.vet_id = specializations.vet_id
+WHERE
+  animal_id != species_id;
+
+--  count
+-- -------
+--      7
+-- (1 row)
+--------------------------------------------------------------------------------------
+/* What specialty should Maisy Smith consider getting? Look for the species she gets the most. */
+SELECT
+  COUNT(DISTINCT species.id) AS different_type,
+  species.name AS species_name
+FROM
+  visits
+  JOIN specializations ON visits.vet_id = specializations.vet_id
+  JOIN vets ON visits.vet_id = vets.id
+  JOIN animals ON visits.animal_id = animals.id
+  JOIN species ON animals.species_id = species.id
+WHERE
+  vets.name = 'Maisy Smith'
+GROUP BY
+  species.name;
```

Comment:
- [Required] I think to get _What specialty should Maisy Smith consider getting? Look for the species she gets the most._  you should have joins between the `animals`, `visits`, `vets`, and the `species` table, the `specializations` table is not neccessary. Currently, your query returns an incorrect result because it includes the specializations table in the joins, please see the query below as an example of how this can be achieved.

```SQL
  SELECT species.name, count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN species ON animals.species_id = species.id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vet.name = 'Maisy Smith'
  GROUP BY species.name
```